lightningd/channel_control.c

```diff
@@ -767,44 +767,34 @@ static void process_check_funding_broadcast(struct bitcoind *bitcoind,
 }
 
 struct command_result *cancel_channel_before_broadcast(struct command *cmd,
-						       const char *buffer,
-						       struct peer *peer,
-						       const jsmntok_t *cidtok)
+						       struct peer *peer)
 {
 	struct channel *cancel_channel;
 	struct channel_to_cancel *cc = tal(cmd, struct channel_to_cancel);
+	struct channel *channel;
 
 	cc->peer = peer->id;
-	if (!cidtok) {
-		struct channel *channel;
-
-		cancel_channel = NULL;
-		list_for_each(&peer->channels, channel, list) {
-			if (cancel_channel) {
-				return command_fail(cmd, LIGHTNINGD,
-						    "Multiple channels:"
-						    " please specify channel_id");
-			}
-			cancel_channel = channel;
-		}
-		if (!cancel_channel)
-			return command_fail(cmd, LIGHTNINGD,
-					    "No channels matching that peer_id");
-		derive_channel_id(&cc->cid,
-				  &cancel_channel->funding_txid,
-				  cancel_channel->funding_outnum);
-	} else {
-		if (!json_tok_channel_id(buffer, cidtok, &cc->cid))
-			return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
-					    "Invalid channel_id parameter.");
-
-		cancel_channel = find_channel_by_id(peer, &cc->cid);
-		if (!cancel_channel)
-			return command_fail(cmd, LIGHTNINGD,
-					    "Channel ID not found: '%.*s'",
-					    cidtok->end - cidtok->start,
-					    buffer + cidtok->start);
+	cancel_channel = NULL;
+	list_for_each(&peer->channels, channel, list) {
+		/* After `fundchannel_complete`, channel is in
+		 * `CHANNELD_AWAITING_LOCKIN` state.
+		 *
+		 * TODO: This assumes only one channel at a time
+		 * can be in this state, which is true at the
+		 * time of this writing, but may change *if* we
+		 * ever implement multiple channels per peer.
+		 */
+		if (channel->state != CHANNELD_AWAITING_LOCKIN)
+			continue;
+		cancel_channel = channel;
+		break;
 	}
+	if (!cancel_channel)
+		return command_fail(cmd, LIGHTNINGD,
+				    "No channels matching that peer_id");
```

Comment:
```suggestion
				    "No channels awaiting lock-in for peer_id %s", ...);
```

There might be a channel, but in the wrong state.

Reply:
Since `fundchannel_cancel` can cancel not just channels awaiting lockin, but also uncommitted channels (i.e. `OPENGIND` state), I reword instead: "No channels being opened or awaiting lock-in for peer_id".